### ISSUE TITLE
logit标准化问题请教

### ISSUE THREAD

**Issue by chaoyangxiao** [sunshangquan/logit-standardardization-KD#24]
作者你好，我想用经典KD和你提出的logit标准化组合一下，训练我的小模型，看看是否能提高病理细粒度图像分类性能。我已经完成了自己的代码，不知是否正确使用了您的方法？

![image](https://github.com/user-attachments/assets/1a4087ea-f2e6-4857-8fe5-1949971fdee3)

`
# 设置损失函数
tmep = 7
hardloss = nn.CrossEntropyLoss()
hardloss = hardloss.cuda()
softloss = DistillKL_logit_stand()
optimizer = torch.optim.Adam(model.parameters(), lr=lr)
scheduler = ReduceLROnPlateau(optimizer, mode='min', factor=0.1, patience=6, verbose=True)
******
for epoch in range(epochs):
    print('-------------------------------第:{}轮训练开始-----------------------------'.format(epoch + 1))
    # 训练开始
    model.train()  # 可有可无
    train_loss = 0  # Initialize training loss
    for data in tqdm(train_dl, desc="Training"):
        start_time2 = time.time()
        imgs, label = data
        imgs, label = imgs.to(device), label.to(device)

        with torch.no_grad():
            teacher_preds = []
            for teacher_model in teacher_models:
                teacher_preds.append(teacher_model(imgs))  
            teacherPred = torch.mean(torch.stack(teacher_preds), dim=0)  

        studentPred= model(imgs)
        stuLoss = hardloss(studentPred, label)

        ##计算蒸馏损失
        distillationLoss = softloss(studentPred,teacherPred,temp)
        # 混合loss
        l = alpha * stuLoss + (1-alpha) * distillationLoss
`

我只是把蒸馏损失换成您提供的损失，不知这样是否正确使用？

**Comment by chaoyangxiao** [comment]
另外，按照论文所描述，总体损失函数的的超参数是否设置为如下较为合理？    
# 混合loss
l = 0.1 * stuLoss + 9 * distillationLoss

**Comment by sunshangquan** [comment]
Hi @chaoyangxiao , 谢谢您的关注，代码应该没有问题，超参数的话 alpha * stuLoss + (1-alpha) * distillationLoss 的权重可以不需要求和为 1，您说的 0.1 和 9 是可以的，也可以尝试1和9，细粒度图像分类的最优权重是否为9可能需要调试一下，也可以尝试 6 或者 12 之类的数值

**Comment by shizhenglove** [comment]
您好我对kd损失权重的设置也有限疑惑，在您的baseline [mdistiller]上针对kd的权重分配为kd0.9 ce0.1，而您的权重设置为kd=9，ce=0.1。并且我在复现过程中res56-20的过程中甚至发现并未添加标准化的同时，于cifar100上的效果就已经达到您论文所展示71.43

![Image](https://github.com/user-attachments/assets/6f723a62-f005-40e4-84cd-e1c8e963c283)

**Comment by ssq-gt** [comment]
KD的数值是沿用了如DKD、ReviewKD等之前论文中的数据，这一个数值可能是由于随机性，您可以再尝试一下其他组的结果
